core/state/statedb.go

```diff
@@ -79,10 +79,10 @@ type StateDB struct {
 
 	// These maps hold the state changes (including the corresponding
 	// original value) that occurred in this **block**.
-	accounts       map[common.Hash][]byte                 // The mutated accounts in 'slim RLP' encoding
-	storages       map[common.Hash]map[common.Hash][]byte // The mutated slots in prefix-zero trimmed rlp format
-	accountsOrigin map[common.Hash][]byte                 // The original value of mutated accounts in 'slim RLP' encoding
-	storagesOrigin map[common.Hash]map[common.Hash][]byte // The original value of mutated slots in prefix-zero trimmed rlp format
+	accounts       map[common.Hash][]byte                    // The mutated accounts in 'slim RLP' encoding
+	storages       map[common.Hash]map[common.Hash][]byte    // The mutated slots in prefix-zero trimmed rlp format
```

Comment:
Shouldn't accounts and storages also be changed in the same vein?

Reply:
Accounts and storages are used by snapshot. Unfortunately, in snapshot, everything is identified by hash.

technically, it's possible to covert from `map[common.Address][]byte` to `map[common.Hash][]byte` inside of snapshot. But i feel it's unworthwhile.